src/proxy.js

```diff
@@ -0,0 +1,278 @@
+                this.password = password;
+                this.availableGroups = groups;
+            } else {
+                throw new Error(this._getMissingParamErrorMgs('password', ENV_VARS.PROXY_PASSWORD));
+            }
+        }
+    }
+
+    /**
+     * Checks the status of apify proxy and
+     * throws an access denied error if the status is not "connected".
+     * Also checks if all proxy groups are available if token is provided.
+     * @returns {Promise<void>}
+     * @private
+     */
+    async _checkAccess() {
+        // Check if user has access to selected proxy groups
+        await this._checkAccessToProxyGroups();
+
+        const url = APIFY_PROXY_STATUS_URL;
+        const proxyUrl = this.getUrl();
+        const countryCode = this.country;
+        const { body } = await requestAsBrowser({ url, proxyUrl, countryCode });
+        const { connected } = JSON.parse(body);
+        if (!connected) this._throwApifyProxyNotConnected();
+    }
+
+    /**
+     * Checks if user has access to each of passed proxy groups
+     * from user object proxy groups and throws error if not
+     * @returns {Promise<void>}
+     * @private
+     */
+    async _checkAccessToProxyGroups() {
+        if (this.groups && this.availableGroups && this.availableGroups.length) {
+            for (const passedGroupName of this.groups) {
+                if (!this.availableGroups.find(availableGroup => availableGroup.name === passedGroupName)) {
+                    this._throwGroupIsNotAvailable(passedGroupName);
+                }
+            }
+        }
+    }
+
+    /**
+     * Validates if parameters groups and country have correct structure
+     * @private
+     */
+    _validateArgumentsStructure(groups, country) {
+        if (groups && groups.length) {
+            for (const group of groups) {
+                if (!APIFY_PROXY_VALUE_REGEX.test(group)) this._throwInvalidProxyValueError(group);
+            }
+        }
+        if (country) {
+            if (!COUNTRY_CODE_REGEX.test(country)) this._throwInvalidCountryCode(country);
+        }
+    }
+
+    /**
+     * Returns missing parameter error message.
+     * @param {string} param
+     * @param {string} env
+     * @return {string} - error message
+     * @private
+     */
+    _getMissingParamErrorMgs(param, env) {
+        return `Apify Proxy ${param} must be provided as parameter or "${env}" environment variable!`;
+    }
+
+    /**
+     * Throws invalid proxy value error
+     * @param {string} param
+     * @private
+     */
+    _throwInvalidProxyValueError(param) {
+        throw new Error(`The "${param}" group option can only contain the following characters: 0-9, a-z, A-Z, ".", "_" and "~"`);
+    }
+
+    /**
+     * Throws invalid country code error
+     * @param {string} code
+     * @private
+     */
+    _throwInvalidCountryCode(code) {
+        throw new Error(`The "${code}" option must be a valid two letter country code according to ISO 3166-1 alpha-2`);
```

Comment:
This error is a bit confusing, e.g. `The "SOMETHING_HERE" option must be a valid two letter country code..."`. How about `The provided country code "SOMETHING_HERE" is not valid. Please use a two-letter code according to...` ?

Reply:
Sure, changed sentence structure.